pkg/kube/kube_handler.go

```diff
@@ -881,56 +861,57 @@ func ValidateOCIAppRepository(appRepo *v1alpha1.AppRepository, cli httpclient.Cl
 	return true, nil
 }
 
-func (r HelmNonOCIValidator) IsValid(cli httpclient.Client) (*ValidationResponse, error) {
+// HttpValidator is an interface for checking the validity of an AppRepo via Http requests.
+type HttpValidator interface {
+	// Validate returns a validation response.
+	Validate(cli httpclient.Client) (*ValidationResponse, error)
+}
+
+// HelmNonOCIValidator is an HttpValidator for non-OCI Helm repositories.
+type HelmNonOCIValidator struct {
+	Req *http.Request
+}
+
+func (r HelmNonOCIValidator) Validate(cli httpclient.Client) (*ValidationResponse, error) {
 
 	response := &ValidationResponse{}
 
-	res, err := cli.Do(r.req)
+	res, err := cli.Do(r.Req)
 	if err != nil {
 		// If the request fail, it's not an internal error
 		return &ValidationResponse{Code: 400, Message: err.Error()}, nil
 	}
-	body, err := ioutil.ReadAll(res.Body)
-	if err != nil {
-		return nil, fmt.Errorf("Unable to parse validation response. Got: %w", err)
-	}
-	response = &ValidationResponse{Code: res.StatusCode, Message: string(body)}
+	response = &ValidationResponse{Code: res.StatusCode, Message: "OK"}
 	if response.Code != 200 {
-		return response, nil
+		body, err := ioutil.ReadAll(res.Body)
+		if err != nil {
+			return nil, fmt.Errorf("Unable to parse validation response. Got: %w", err)
+		}
+		response.Message = string(body)
 	}
 
 	return response, nil
 }
 
-func (r HelmNonOCIValidator) getReq() *http.Request {
-	return r.req
+type HelmOCIValidator struct {
+	AppRepo *v1alpha1.AppRepository
 }
 
-func (r HelmNonOCIValidator) getAppRepo() *v1alpha1.AppRepository {
-	return nil
-}
-func (r HelmOCIValidator) IsValid(cli httpclient.Client) (*ValidationResponse, error) {
+func (r HelmOCIValidator) Validate(cli httpclient.Client) (*ValidationResponse, error) {
 
 	var response *ValidationResponse
-	response = &ValidationResponse{Code: 200, Message: ""}
+	response = &ValidationResponse{Code: 200, Message: "OK"}
 
-	isValidRepo, err := ValidateOCIAppRepository(r.appRepo, cli)
-	if !isValidRepo {
+	// If there was an error validating the OCI repository, it's not an internal error.
+	isValidRepo, err := ValidateOCIAppRepository(r.AppRepo, cli)
+	if err != nil || !isValidRepo {
 		response = &ValidationResponse{Code: 400, Message: err.Error()}
 	}
-	return response, err
```

Comment:
Generally in Go, if you have an error, you return `nil, err`, if you don't have an error, you return `response, nil`, keeping them mutually exclusive. Again, the test brought this out so that we now return a validation response here. We could potentially remove the error from the signature, if it was only the OCI validator, but the non-OCI validator still has cases where it returns an error and I didn't want to change that here.